vsce/server/src/server.ts

```diff
@@ -286,7 +288,34 @@ async function validateTextDocument(textDocument: TextDocument): Promise<void> {
 	const reachPath = (exeLoc == '' || exeLoc == './reach')
 		? path.join(workspaceFolder, "reach")
 		: exeLoc;
+
+	if (theCompilerIsCompiling) {
+		console.debug(
+			"Compilation is already in process! Returning...",
+			new Date().toLocaleTimeString()
+		);
+		return;
+	}
+
+	theCompilerIsCompiling = true;
+
+	console.debug(
+		"Starting compilation at",
+		new Date().toLocaleTimeString()
+	);
 	await exec("cd " + tempFolder + " && " + reachPath + " compile " + REACH_TEMP_FILE_NAME + " --error-format-json", (error: { message: any; }, stdout: any, stderr: any) => {
+		// This callback function should execute exactly
+		// when this compilation command has finished.
+		// "child_process.exec(): spawns a shell...
+		// passing the stdout and stderr to a callback
+		// function when complete". See
+		// https://nodejs.org/api/child_process.html#child-process
+		console.debug(
+			"Compilation should now have finished.",
+			new Date().toLocaleTimeString()
+		);
+		theCompilerIsCompiling = false;
```

Comment:
And then here, do `if ( doAnotherCompile ) { goAgain(); }`

Reply:
Thank you for the suggestion, and specifically for pointing out the line to which to add the suggestion: that's helpful.

1304276 should address this.